xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java

```diff
@@ -894,4 +895,20 @@ public Template getTemplate(String templateName)
 
         return template;
     }
+
+    /**
+     * Create a DefaultTemplate with the given content and the given author.
+     *
+     * @param content the template content
+     * @param author the template author
+     * @return the template
+     */
+    public Template createStringTemplate(String content, DocumentReference author)
+    {
+        DefaultTemplate newTemplate = new DefaultTemplate(null);
```

Comment:
> Unfortunately, I also found out that adding the new method createStringTemplate to the TemplateManager interface would make RevAPI scream during maven build (I forgot that TemplateManager was a public API).

Yes this is always the case in interface. reapi is not really complaining because you are modifying an API but because older implementations of TemplateManager will not have implemented this new method.

Fortunately Java 8 introduced a solution for this use case. You can use default keyword in front of an interface method to provide a default implementation in case none is found in the actual class implementing the interface.